### PR TITLE
Oleksa

### PR DESCRIPTION
**Pull request by Askeloo**
added messages cache to show on GUI and added register to GUI

### CODE REVIEW

File: Peer/GUI/guimanager.cpp [hunk 1]
```diff
@@ -62,31 +62,53 @@ void GUIManager::deleteFriend(FriendItem* friend_to_delete) {
...
   message_model_.AddMessageToList(new_message);
 }
 
-void GUIManager::LoadMessages(unsigned friend_id) {
+void GUIManager::ShowMessages(unsigned friend_id) {
   if (friend_id) {
     message_model_.RemoveAllMessagesFromList();
 
     QString data, time, date;
     int owner_id;
```
Comment (FedorenkoAndriy) [inline Peer/GUI/guimanager.cpp]: You are no longer using this variable.

File: Peer/GUI/guimanager.cpp [hunk 2]
```diff
@@ -62,31 +62,53 @@ void GUIManager::deleteFriend(FriendItem* friend_to_delete) {
...
     MessageItem* new_message;
```
Comment (FedorenkoAndriy) [inline Peer/GUI/guimanager.cpp]: The same

File: Peer/GUI/guimanager.cpp [hunk 3]
```diff
@@ -62,31 +62,53 @@ void GUIManager::deleteFriend(FriendItem* friend_to_delete) {
...
+
+    for (Message* msg : messages_cache_[friend_id]) {
+      new_message = new MessageItem(msg);
       message_model_.AddMessageToList(new_message);
     }
   }
 }
 
+void GUIManager::LoadAllMessages(unsigned friend_id) {
+  Message* temp;
```
Comment (FedorenkoAndriy) [inline Peer/GUI/guimanager.cpp]: Looks like this variable is redundant

File: Peer/GUI/guimanager.cpp (hunk 1 above)
Comment (Askeloo) [reply Peer/GUI/guimanager.cpp]: it is already fixed)

**Pull request merged**